src/OpenTelemetry.Exporter.Geneva/GenevaExporterHelperExtensions.cs

```diff
@@ -26,34 +27,31 @@ public static TracerProviderBuilder AddGenevaTraceExporter(this TracerProviderBu
     {
         Guard.ThrowIfNull(builder);
 
-        if (builder is IDeferredTracerProviderBuilder deferredTracerProviderBuilder)
+        return builder.AddProcessor(sp =>
         {
-            return deferredTracerProviderBuilder.Configure((sp, builder) =>
-            {
-                AddGenevaTraceExporter(builder, sp.GetOptions<GenevaExporterOptions>(), configure);
-            });
-        }
+            var exporterOptions = sp.GetOptions<GenevaExporterOptions>();
```

Comment:
@CodeBlanch I have updated the extension methods for both GenevaTraceExporter and GenevaMetricExporter to use the newer DI APIs. Could you please review these changes?

Reply:
This looks good.

There are a couple things missing, but up to you if you want to shore it up :smile: I'll just drop details here you can tackle now or later if you want.

   1) Named options support is missing. Probably doesn't seem important, but I think because all signals share an options class it actually is very useful for GenevaExporter (& OtlpExporter).
   2) Options API integration won't work totally as expected.

I think 1 is pretty self-explanatory so I won't go into it but 2 maybe not as clear; here is an example:

```csharp
appBuilder.Services.PostConfigure<GenevaExporterOptions>(o =>
{
    o.ConnectionString = "[connection_string_2]";
});

appBuilder.Services.AddOpenTelemetry().WithTracing(builder =>
   builder.AddGenevaTraceExporter(o => {
      o.ConnectionString = "[connection_string_1]";
   }));
```

In that snippet the user probably expected the "PostConfigure" delegate to run last. But the way the code is written, the inline one will win.

The reason for that is on line 40 below where we do `configure?.Invoke(options);` that is executing it inline and not participating correctly in the Options API world.

Reply:
Thanks for sharing this insight! I'll address these in a different PR.